docs/src/user_docs/assembly/code_organization.md

```diff
@@ -51,12 +44,63 @@ begin
 end
 ```
 
-In addition to local procedures defined within a program, Miden assembly programs can invoke procedures defined in the standard library as described [here](../stdlib/main.md).
+### Modules
+A *module* consists of one or more procedures. There are two types of modules: *library modules* and *executable modules* (also called *programs*).
 
-### Comments
-Miden assembly allows annotating code with simple comments. Currently, the only supported type of comments is a single-line comment which starts with a `#` (pound) character. For example:
+#### Library modules
+Library modules can contain zero or more internal procedures and one or more exported procedures. For example, the following module defines one internal procedure (defined with `proc` instruction) and one exported procedure (defined with `export` instruction):
```

Comment:
nit: If it's a requirement for creating these, then I would replace "can contain" with "contain"